zipkin-lens/src/components/DependenciesPage/DependenciesPage.jsx

```diff
@@ -0,0 +1,245 @@
+import Box from '@material-ui/core/Box';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+import DependenciesGraph from './DependenciesGraph';
+import DependenciesPageHeader from './DependenciesPageHeader';
+import NodeDetail from './NodeDetail';
+import * as dependenciesActionCreators from '../../actions/dependencies-action';
+import Graph from '../../util/dependencies-graph';
+import ExplainBox from './ExplainBox';
+import { buildQueryParameters } from '../../util/api';
+
+const useStyles = makeStyles(theme => ({
+  header: {
+    borderBottom: `1px solid ${theme.palette.grey[300]}`,
+  },
+  timePicker: {
+    marginLeft: theme.spacing(1),
+    marginRight: theme.spacing(1),
+  },
+  timePickerInput: {
+    fontSize: '1rem',
+    height: '1.6rem',
+    padding: '0.4rem 0.6rem',
+  },
+  findButton: {
+    fontSize: '1.2rem',
+    padding: theme.spacing(1),
+    minWidth: '10px',
+  },
+  loadingIndicatorWrapper: {
+    width: '100%',
+    height: '100%',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  explainBoxWrapper: {
+    width: '100%',
+    height: '100%',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  content: {
+    width: '100%',
+    height: '100%',
+    display: 'flex',
+    backgroundColor: theme.palette.background.paper,
+  },
+  graphWrapper: {
+    height: '100%',
+  },
+  nodeDetailWrapper: {
+    width: '30%',
+    height: '100%',
+  },
+}));
+
+const propTypes = {
+  isLoading: PropTypes.bool.isRequired,
+  dependencies: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+  fetchDependencies: PropTypes.func.isRequired,
+  clearDependencies: PropTypes.func.isRequired,
+  location: PropTypes.shape({ search: PropTypes.string.isRequired }).isRequired,
+  history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
+};
+
+export const DependenciesPageImpl = React.memo(({
+  isLoading,
+  dependencies,
+  fetchDependencies,
+  clearDependencies,
+  location,
+  history,
+}) => {
+  const classes = useStyles();
+  const graph = useMemo(() => new Graph(dependencies), [dependencies]);
+  const isGraphExists = graph.allNodes().length !== 0;
+  const [nodeName, setNodeName] = useState(null);
+  const [timeRange, setTimeRange] = useState({
+    startTime: moment().subtract(1, 'days'),
+    endTime: moment(),
+  });
+  const targetEdges = useMemo(
+    () => (nodeName ? graph.getTargetEdges(nodeName) : []),
+    [nodeName, graph],
+  );
+  const sourceEdges = useMemo(
+    () => (nodeName ? graph.getSourceEdges(nodeName) : []),
+    [nodeName, graph],
+  );
+
+  const handleStartTimeChange = useCallback((startTime) => {
+    setTimeRange({ ...timeRange, startTime });
+  }, [timeRange]);
+
+  const handleEndTimeChange = useCallback((endTime) => {
+    setTimeRange({ ...timeRange, endTime });
+  }, [timeRange]);
+
+  const handleFindButtonClick = useCallback(() => {
+    const startTs = timeRange.startTime.valueOf();
```

Comment:
the parameters are endTs and lookback :) https://zipkin.io/zipkin-api/#/default/get_dependencies

Reply:
oops!!